enclave-runtime/src/top_pool_execution.rs

```diff
@@ -178,3 +190,51 @@ where
 
 	Ok(())
 }
+
+/// Executes aura for the given `slot`
+fn exec_aura_on_slot<Authority, PB, SB, OCallApi, LightValidator, PEnvironment, ExtrinsicsFactory>(
+	slot: SlotInfo<PB>,
+	authority: Authority,
+	validator: &mut LightValidator,
+	extrinsics_factory: &ExtrinsicsFactory,
+	ocall_api: OCallApi,
+	proposer_environment: PEnvironment,
+	shards: Vec<ShardIdentifierFor<SB>>,
+) -> Result<()>
+where
```

Comment:
Moved this function here from `sidechain_impl.rs`. It does a lot of component construction and handles the top-level workflow for executing aura. So for the moment, it remains in `enclave-runtime`.